GrayBMP.cs

```diff
@@ -126,6 +126,41 @@ public void Dirty ()
       End ();
    }
 
+   public void DrawThickLine (int x1, int y1, int x2, int y2, int width, int gray) {

```

Comment:
Your routine will draw a line with width = **width * 2** actually. Either scale the parameter inside the routine (preferred), or change the parameter name to halfWidth and make it a float / double.

Reply:
Yes. Passing **mScale / 2** hid this error earlier. Thanks.